lib/project_types/theme/messages/messages.rb

```diff
@@ -32,6 +32,26 @@ module Messages
             APP
           },
         },
+        push: {
+          error: {
+            pushing_error: "Theme files couldn't be pushed to Shopify",
+            removing_error: "Theme files couldn't be removed from Shopify",
+          },
+          help: <<~HELP,
+            {{command:%s push}}: Replaces theme files on Shopify with those in your directory. If filenames are provided, only those files will be replaced; otherwise, the whole theme is replaced.
+              Usage: {{command:%s push}}
+              Options:
+                {{command:--remove}} Removes, rather than replaces, both local and Shopify copies of theme files. At least one file must be specified.
+                {{command:--allow-live}} Allows Shopify App CLI to replace a file on the currently live theme.
```

Comment:
I see "allow-live" is carried over from the existing Theme Kit commands. I do wonder if we could come up with a better option name for this though — "allow live" doesn't seem to describe fully what the option enables. Any thoughts @gfscott?

Reply:
_If_ I were designing this from scratch — a big if! — I'd create a flag to specify which environment or context to push to:

```console
# pushing to the non-production context would be the default, so these two commands would be equivalent:
$ shopify push
$ shopify push --environment staging

# pushing to the live production site:
$ shopify push --environment production
```

This makes it possible to extend the option in future with other arbitrarily-named environments (`dev`, `canary`, `us-east-1`).

I'm totally agnostic on what the flag itself should be; `environment` may not be appropriate and I'm not sure "staging" and "production" are quite accurate either, but you could substitute other names and the principle still holds.

You asked lol 😅

Reply:
There will be an `--env` flag added later to specify which environment to push to. For now, `--allow-live` simply lets the user change the live theme (which could potentially be risky).